### PR TITLE
Isomorphic rendering support.

### PR DESCRIPTION
**Pull request by hanjukim**
* Move new window.Image to componentDidMount for isomorphic (universal) rendering.
* Update React modules to ^15.
* Remove development files from package to reduce size.

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,2 +1,8 @@
 node_modules
-npm-debug.log
\ No newline at end of file
+npm-debug.log
+src
```
Comment (conorhastings) [inline .npmignore]: i'd prefer to leave this one, find myself looking at src in node_modules directory quite often

File: package.json [hunk 2]
```diff
@@ -4,12 +4,12 @@
   "description": "if your image doesn't exist, fallback onto another provided image.",
   "main": "lib/index.js",
   "scripts": {
-    "build": "babel src --out-dir lib",
-    "prepublish": "npm run build",
+    "build": "./node_modules/babel-cli/bin/babel.js src --out-dir lib",
```
Comment (conorhastings) [inline package.json]: why this change?

File: package.json [hunk 3]
```diff
@@ -4,12 +4,12 @@
...
+    "prepublishOnly": "npm run build",
     "lint": "./node_modules/.bin/eslint ./src",
     "test": "npm run build && npm run lint &&  browserify test/index.js | tap-closer | smokestack -b firefox | tap-status",
     "test-debug": "npm run build && npm run lint &&  browserify test/index.js | smokestack | tap-status",
-    "watch": "babel src --watch --out-dir lib"
+    "watch": "./node_modules/babel-cli/bin/babel.js src --watch --out-dir lib"
```
Comment (conorhastings) [inline package.json]: why this change?

File: package.json (hunk 2 above)
Comment (hanjukim) [inline package.json]: That's because CircleCI build fails. I cannot reproduce in my local environment. I'm suspecting there's some kind of cache problem.

Failed results:
https://circleci.com/gh/socialtables/react-image-fallback/51
https://circleci.com/gh/socialtables/react-image-fallback/54

Fixed result:
https://circleci.com/gh/socialtables/react-image-fallback/55

File: package.json (hunk 3 above)
Comment (hanjukim) [inline package.json]: Same as above reason.

**Comment by conorhastings** [comment]
@hanjukim published as `4.0.0`

**Comment by conorhastings** [comment]
sorry, was not running`npm@4` , `4.0.1` should have your changes
